### PR TITLE
Added Employee List TestCases

### PR DESCRIPTION
**Pull request by shubhammehta04**

### CODE REVIEW

File: cypress/e2e/pages/EmployeeListPage.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  cy.log("Entered Invalid Name to Search")
+    }  
+
+validateNoRecordsAppear(){
+  this.noRecordAvailable.should('have.text', "No Records Available");
+  cy.log("No Records Appear")
+
+}    
+
+countTotalEmployees(){
```
Comment (Sandy-N) [inline cypress/e2e/pages/EmployeeListPage.js:29]: correct this

File: cypress/e2e/pages/EmployeeListPage.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  this.employeeCount.invoke('text')
+  .then((text) => {
+  const employeeCount = parseInt(text.replace('Total Employees : ', '').trim());
+  expect(employeeCount).to.equal(1);  
+  cy.log(`Total number of employees: ${employeeCount}`);
+        });
+      }
+
+selectSearchedUser(){
```
Comment (Sandy-N) [inline cypress/e2e/pages/EmployeeListPage.js:38]: fix this

File: cypress/e2e/pages/UserDashboard.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  this.firstName.should('not.be.visible');
+  cy.log("Accordion Is Collapsed")
+ } 
+
+clickOnEditButtonBasicInfo(){
+  this.editBasicInfo.should('be.visible').click();
+  cy.log("Clicked on the Edit button Under Basic Info Accordion");
+}  
+
+updateNameFields(firstNameText, lastNameText, middleNameText){
```
Comment (Sandy-N) [inline cypress/e2e/pages/UserDashboard.js:66]: update this

File: cypress/e2e/pages/UserDashboard.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  cy.log("Clicked on the Close Button under Basic Info Accordion");
+} 
+
+clickOnUpdateButton(){
+  this.updateButton.should('be.visible').click();
+  cy.log("Clicked on the Update Button under Basic Info Accordion");
+  Loaders.threeDotLoading.should('not.exist');
+}   
+
+verifyDataForAllFields(firstNameTxt, middleNameTxt, lastNameTxt, employeeId, emailID){
```
Comment (Sandy-N) [inline cypress/e2e/pages/UserDashboard.js:88]: update this

File: cypress/e2e/pages/UserDashboard.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
+import BasePage from "./BasePage";
+import Loaders from "../components/Loaders";
+
+class UserDashboard extends BasePage{
+    
+// Locators    
+get workExperience() { return cy.get("h2[id='heading4'] button[type='button']")}
```
Comment (Sandy-N) [inline cypress/e2e/pages/UserDashboard.js:7]: change all locators

File: cypress/e2e/pages/UserDashboard.js [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+  cy.log("Clicked on the Close Button under Basic Info Accordion");
+} 
+
+clickOnUpdateButton(){
+  this.updateButton.should('be.visible').click();
+  cy.log("Clicked on the Update Button under Basic Info Accordion");
+  Loaders.threeDotLoading.should('not.exist');
+}   
+
+verifyFirstName(firstNameTxt){
```
Comment (Sandy-N) [inline cypress/e2e/pages/UserDashboard.js:99]: delete all these verify methods

File: cypress/e2e/pages/UserDashboard.js [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+  cy.log("Updated Middle Name Text");
+}
+
+updateLastName(lastNameText){
+  this.editLastName.clear().type(lastNameText).should('have.value',lastNameText)
+  cy.log("Updated Last Name Text");
+}
+
+clickOnCloseButton(){
+  this.closeButton.should('be.visible').click();
```
Comment (Sandy-N) [inline cypress/e2e/pages/UserDashboard.js:89]: remove additional assertion

File: cypress/e2e/pages/UserDashboard.js [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+  cy.log("Accordion Is Collapsed")
+ } 
+
+clickOnEditButtonBasicInfo(){
+  this.editBasicInfo.should('be.visible').click();
+  cy.log("Clicked on the Edit button Under Basic Info Accordion");
+}  
+
+updateFirstName(firstNameText){
+  this.editFirstName.clear().type(firstNameText).should('have.value',firstNameText)
```
Comment (Sandy-N) [inline cypress/e2e/pages/UserDashboard.js:74]: check clear()

File: cypress/e2e/tests/EmployeeList.cy.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+import UserDashboard from "../pages/UserDashboard";
+
+describe("Employee Onboard Tests", () => {
+
+    it("HRMIS_1: Verify that 'Employee List' page opens up, when user clicks on 'Employee Management'.", () => {
+        cy.login();
+        sideBar.navigateTo("Employee Management", "Employees List");
+
+        // Verify that information message, when user enters invalid data in 'Search By Name' field
+        EmployeeListPage.enterInvalidNameToSearch('Tests');
```
Comment (Sandy-N) [inline cypress/e2e/tests/EmployeeList.cy.js:12]: make same method for 2 uses

File: cypress/e2e/tests/EmployeeList.cy.js [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+        EmployeeListPage.validateNoRecordsAppear();
+
+        // Verify that information message, when user enters valid data in 'Search By Name' field
+        EmployeeListPage.enterValidNameToSearch('Aut');
```
Comment (Sandy-N) [inline cypress/e2e/tests/EmployeeList.cy.js:16]: full name

File: cypress/e2e/tests/EmployeeList.cy.js [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+        EmployeeListPage.countTotalEmployees(1);
+
+        // Verify that the information message appears for the selected department with no records.
+        EmployeeListPage.selectSearchedUser();
+        UserDashboard.clickOnWorkExperience();
+        UserDashboard.validateNoRecordsAppear();
+        
+        // Verify that 'Basic Info' accordion gets expanded, when user clicks 'Basic Info' accordion.
+        UserDashboard.clickOnBasicInfo();
```
Comment (Sandy-N) [inline cypress/e2e/tests/EmployeeList.cy.js:25]: update with should have

File: cypress/e2e/pages/UserDashboard.js (hunk 6 above)
Comment (shubhammehta04) [reply cypress/e2e/pages/UserDashboard.js:74]: Done

File: cypress/e2e/tests/EmployeeList.cy.js (hunk 11 above)
Comment (shubhammehta04) [reply cypress/e2e/tests/EmployeeList.cy.js:26]: Done

File: cypress/e2e/tests/EmployeeList.cy.js (hunk 10 above)
Comment (shubhammehta04) [reply cypress/e2e/tests/EmployeeList.cy.js]: Done

File: cypress/e2e/tests/EmployeeList.cy.js (hunk 9 above)
Comment (shubhammehta04) [reply cypress/e2e/tests/EmployeeList.cy.js]: Done

File: cypress/e2e/pages/UserDashboard.js (hunk 8 above)
Comment (shubhammehta04) [reply cypress/e2e/pages/UserDashboard.js]: Done

File: cypress/e2e/pages/UserDashboard.js (hunk 7 above)
Comment (shubhammehta04) [reply cypress/e2e/pages/UserDashboard.js]: Done

File: cypress/e2e/pages/UserDashboard.js (hunk 5 above)
Comment (shubhammehta04) [reply cypress/e2e/pages/UserDashboard.js:7]: Done

File: cypress/e2e/pages/EmployeeListPage.js (hunk 1 above)
Comment (Sandy-N) [reply cypress/e2e/pages/EmployeeListPage.js]: return number of employees and assert it in test class

**Pull request merged**